pkg/signature/kms/cliplugin/README.md

```diff
@@ -0,0 +1,91 @@
+# CLI Plugin
+
+This is a package and module for using custom KMS plugins as separate executables.
+It is intended to be used by cosign, but you may use this in your own programs that import sigstore.
+
+## Design
+
+We follow [kubectl's style](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/#writing-kubectl-plugins) of plugins. Any language that can create CLI programs can be used to make a plugin.
+
+### Usage
+
+Plugins are separate programs on your system's PATH, named in the scheme `sigstore-kms-[name]`, like `sigstore-kms-my-hsm`. They can be invoked with cosign like `cosign [sub-command] --key "my-hsm://my-key-id" ...
+`
+
+### Protocol
+
+The main program will invoke the program with these specifications:
+
+* stdin
+  * Data to be signed or verified.
+* arg 1
+  * A number identifying the version of this protocol.
+    * In the future we may change this protocol, either the encoding or the formatting of argument and return values.
+    Protocol changes will result in a major-version bump for the library.
+    When changing the protocol, new versions of sigstore will not maintain backwards compatibility with
+    previous protocol versions. If a plugin author wishes, they may branch their plugin program’s behaviour
+    to be compatible with multiple versions of the protocol, or multiple major versions of sigstore and cosign.
+* arg 2
+  * JSON of initialization options and method arguments.
```

Comment:
why could these not be combined?

Reply:
It's possible we use a very different protocol than the current version. In order to keep things simple for future plugin programs, I think we should keep this first argument simple to parse.